### PR TITLE
Fix flaky unified scheduler pool tests

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

some unified scheduler pool tests are flaky by nature.

#### Summary of Changes

Originally, i wanted to improve sleepless_testing or rewrite the tests to make them less racy... however, it turned out it's rather involved and not worth all the trouble....

So, just make tests run longer by shamelessly increasing some constant test knobs, along with some execuse commentary...

### CODE REVIEW

File: unified-scheduler-pool/src/lib.rs [hunk 1]
```diff
@@ -2080,7 +2094,8 @@ mod tests {
         sleepless_testing::at(TestCheckPoint::BeforeThreadManagerDrop);
         drop::<PooledScheduler<_>>(scheduler);
         sleepless_testing::at(TestCheckPoint::AfterSchedulerThreadAborted);
-        assert!(*TASK_COUNT.lock().unwrap() < 10);
+        // All of handler threads should have been aborted before processing MAX_TASK_COUNT tasks.
+        assert!(*TASK_COUNT.lock().unwrap() < MAX_TASK_COUNT);
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:2098]: https://buildkite.com/anza/agave/builds/9084#019132e7-6473-47b7-91a8-464c9940f801

File: unified-scheduler-pool/src/lib.rs [hunk 2]
```diff
@@ -1575,6 +1575,13 @@ mod tests {
...
         // See the old (= idle) scheduler gone only after solScCleaner did its job...
         sleepless_testing::at(&TestCheckPoint::AfterIdleSchedulerCleaned);
+
+        // The following assertion is racy.
+        //
+        // We need to make sure new_scheduler isn't treated as idle up to now since being returned
+        // to the pool after sleep(SHORTENED_TIMEOUT_DURATION * 10).
+        // Removing only old_scheduler is the expected behavior. So, make
+        // SHORTENED_TIMEOUT_DURATION rather long...
         assert_eq!(pool_raw.scheduler_inners.lock().unwrap().len(), 1);
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:1585]: After looping those tests awhile, this found to be racy as well.

File: unified-scheduler-pool/src/lib.rs [hunk 3]
```diff
@@ -1575,6 +1575,13 @@ mod tests {
 
         // See the old (= idle) scheduler gone only after solScCleaner did its job...
         sleepless_testing::at(&TestCheckPoint::AfterIdleSchedulerCleaned);
+
+        // The following assertion is racy.
+        //
+        // We need to make sure new_scheduler isn't treated as idle up to now since being returned
+        // to the pool after sleep(SHORTENED_TIMEOUT_DURATION * 10).
```
Comment (apfitzge) [inline unified-scheduler-pool/src/lib.rs]: is `SHORTENED_TIMEOUT_DURATION` the same as `SHORTENED_MAX_POOLING_DURATION`?

Comment (ryoqun) [reply unified-scheduler-pool/src/lib.rs]: oops, nice catch of bad copy paste... 0997fab5a46ac683f215383f0de8e7a2afba7426

**Review by apfitzge — APPROVED** [review]

**Pull request merged**